python/ray/util/metrics.py

```diff
@@ -0,0 +1,182 @@
+from typing import Dict, Any, List, Optional
+
+from ray._raylet import (
+    Count as CythonCount,
+    Histogram as CythonHistogram,
+    Gauge as CythonGauge,
+)  # noqa: E402
+
+
+class Metric:
+    """The parent class of custom metrics.
+
+    Ray's custom metrics APIs are rooted from this class and sharing
+    the same public methods.
+
+    Recommended metric name pattern : ray.{component_name}.{module_name}, and
+    name format must be in [0-9a-zA-Z].
+    """
+
+    def __init__(self,
+                 name: str,
+                 description: str = "",
+                 tags: Dict[str, Optional[str]] = None):
```

Comment:
can we support tuple of str as well? instead of `{"label_a": None, "label_2": None}`, can we just do `tags=("label_a", "label_2")`. This is how it works in prometheus and a lot more concise.

Reply:
The motivation of this change was from this issue https://github.com/ray-project/ray/issues/10184

=> Provide some way to specify default tags for metrics. 

So, when the value is None, there's no default, and when the value is given, they will be default. I can change the API to require both fields (which was my original implementation). In this case, the API will look like

```
self,
name
description
tag_keys: Tuple[str]
default_tags: Dict[str]
```

Reply:
I'm fine with tag_keys and default_tags. it's a lot clearer